### PR TITLE
INTYGFV-15037: No conversion of fieldNames (in validationMessages) fo…

### PR DESCRIPTION
**Pull request by mww-sopra-steria**
…r other types than lisjp and ag7804.

### CODE REVIEW

**Review by ChrilleSigerhed — APPROVED** [review]
Awsome, great work!

File: sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java [hunk 1]
```diff
@@ -149,22 +149,34 @@ private static int getElementId(String questionId) {
...
-        for (String questionId : questionIds) {
-            if (certificate.getData().get(questionId) != null) {
-                final var terminalCauseOfDeath = (CertificateDataValueCauseOfDeath) certificate.getData().get(questionId)
-                    .getValue();
-                final var description = terminalCauseOfDeath.getDescription().getText();
-                final var debut = terminalCauseOfDeath.getDebut().getDate() != null
-                    ? new InternalDate(terminalCauseOfDeath.getDebut().getDate()) : null;
-                final var specifikation = terminalCauseOfDeath.getSpecification().getCode() != null
-                    ? Specifikation.fromValue(terminalCauseOfDeath.getSpecification().getCode()) : null;
+        final List<Dodsorsak> dodsorsakList = questionIds.stream()
```
Comment (IamFriberg) [inline sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java:153]: Is it possible to break this into smaller chunks or private methods to make this more descriptive?

**Review by IamFriberg — CHANGES_REQUESTED** [review]

File: sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java [hunk 2]
```diff
@@ -149,22 +149,34 @@ private static int getElementId(String questionId) {
...
+            .filter(questionId -> certificate.getData().containsKey(questionId))
+            .map(questionId -> (CertificateDataValueCauseOfDeath) certificate.getData().get(questionId).getValue())
```
[the remark this answers is not in the document]
Comment (mww-sopra-steria) [reply sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java:155]: So we don't need to do null-checks if the question doesn't exist or not.

File: sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java (hunk 1 above)
Comment (mww-sopra-steria) [reply sos/doi/src/main/java/se/inera/intyg/common/doi/v1/model/converter/certificate/question/QuestionTerminalDodsorsakFoljdAv.java:153]: Sure, just used the original code. Done!

**Review by IamFriberg — APPROVED** [review]
Snyggt!

**Pull request merged**